src/app/store/sagas/window/createActionChannel.ts

```diff
@@ -0,0 +1,48 @@
+import Logger from 'app/utils/Logger';
+import { eventChannel } from 'redux-saga';
+import { StandardAction } from 'app/store/types';
+import isStandardAction from 'app/store/isStandardAction';
+
+export type StandardActionMatcher = (action: StandardAction) => boolean;
+
+export const createMessageHandler = (
+  emit: (action: StandardAction) => void
+) => (targetOrigin = '*') => (actionMatcher: StandardActionMatcher) => (
+  event: MessageEvent
+) => {
+  console.log('targetOrigin', targetOrigin, event.origin);
+  if (isStandardAction(event.data) && targetOrigin === event.origin) {
+    if (actionMatcher(event.data)) {
+      emit(event.data);
+    } else {
+      if (process.env.NODE_ENV !== 'production') {
+        Logger.debug("Didn't match:", event.data);
+      }
+    }
+  } else {
+    if (process.env.NODE_ENV !== 'production') {
+      Logger.debug('Invalid action:', event.data);
+    }
+  }
+};
+
+const createActionChannel = (
+  window: Window,
+  targetOrigin = '*',
```

Comment:
Since we match`targetOrigin === event.origin`, Will `targetOrigin` of `*` ever match anything ?